app/controllers/pet_applications_controller.rb

```diff
@@ -0,0 +1,13 @@
+class PetApplicationsController < ApplicationController
+  def create
```

Comment:
Nice work pulling this functionality into a 3rd Joins controller! This is the most RESTful option.